tree: cast TIMESTAMP to TEXT without timezone data

otan opened the pull request:
This better matches postgres. I'm guessing it used to be done this way
to match the CLI output (which we can't fix without changing lib/pq),
but this affects output results from ORM tests. Note that CLI output is
not affected unless TIMESTAMP is cast to text.

Release note (sql change): Casting TIMESTAMP types to TEXT related types
now omits the timezone component. For example, '2001-12-15
15:14:13'::TIMESTAMP will now format as '2001-12-15 15:14:13' instead of
'2001-12-15 15:14:13+00:00'.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51692)
<!-- Reviewable:end -->

knz commented:
Can you explain in the release note (perhaps anotherone with category 'cli change') what the expected change is in the CLI shell.

otan commented:
No cli change. They will appear the same unless you cast it to text.

On Wed, 22 Jul 2020, 2:48 am kena, <notifications@github.com> wrote:

> Can you explain in the release note (perhaps anotherone with category 'cli
> change') what the expected change is in the CLI shell.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/51692#issuecomment-662356992>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA32FQ3ZCGMUGDEHQQPBYNTR42YWRANCNFSM4PEH2E3Q>
> .
>

otan committed:
tree: cast TIMESTAMP to TEXT without timezone data

This better matches postgres. I'm guessing it used to be done this way
to match the CLI output (which we can't fix without changing lib/pq),
but this affects output results from ORM tests. Note that CLI output is
not affected unless TIMESTAMP is cast to text.

Release note (sql change): Casting TIMESTAMP types to TEXT related types
now omits the timezone component. For example, '2001-12-15
15:14:13'::TIMESTAMP will now format as '2001-12-15 15:14:13' instead of
'2001-12-15 15:14:13+00:00'.

otan commented:
TFTR!

bors r=rohany,adityamaru
